rectools/model_selection/cross_validate.py

```diff
@@ -137,7 +204,35 @@ def cross_validate(  # pylint: disable=too-many-locals
         else:
             item_ids_to_recommend = None
 
+        if ref_models is not None:
+            ref_reco = _get_ref_reco(
+                ref_models=ref_models,
+                models=models,
+                test_users=test_users,
+                fold_dataset=fold_dataset,
+                k=k,
+                filter_viewed=filter_viewed,
+                internal_item_ids_to_recommend=item_ids_to_recommend,
+            )
+            models_ref_reco = {model_name: None if model_name in ref_models else ref_reco for model_name in models}
+
+            metrics_without_intersection = {
+                metric_name: metric
+                for metric_name, metric in metrics.items()
+                if not isinstance(metric, IntersectionMetric)
+            }
+            models_metrics = {
```

Comment:
The problem in logic is here. Imagine the case when we have multiple `ref_models` and we have `validate_ref_models` set to True.
In this case we need to calculate all metrics (including intersection ones) for all models.
The only thing we can drop is calculating model intersection with itself (we can return `1` for this case)